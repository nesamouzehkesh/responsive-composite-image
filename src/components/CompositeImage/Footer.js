import React, {Component} from 'react';
import styled from 'styled-components';

const StyledFooter = styled.div`
  display: flex;
  align-items: center;
  width:100%;
  height:30%;
  background-color: rgba(0, 0, 0, ${props => props.opacity});
  position:absolute;
  bottom:0px;
  z-index: 1;
`;

const StyledTitle = styled.span`
  font-size: 20px;
  font-family: Helvetica, Arial, sans-serif;
  font-weight: bold;
  color: #fff;
  /* Tablets, Ipads (portrait) */
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 22px;
  }
  /* Laptops, Desktops */
  @media all and (min-width: 1025px) and (max-width: 1280px) {
    font-size: 15px;
  }
  /* Desktops */
  @media only screen and (min-width: 1281px) {
    font-size: 15px;
  }
`;

const StyledLogo = styled.img`
  margin: 10px;
  width: 40px;
  /* Tablets, Ipads (portrait) */
  @media (min-width: 768px) and (max-width: 1024px) {
    width: 58px;
  }
  /* Laptops, Desktops */
  @media all and (min-width: 1025px) and (max-width: 1280px) {
    width: 40px;
  }
  /* Desktops */
  @media only screen and (min-width: 1281px) {
    width: 40px;
  }  
`;

function Footer({logo, title, opacity}) {
  return (
    <StyledFooter opacity={opacity}>
      <StyledLogo src={logo} title={title} />
      <StyledTitle>{title}</StyledTitle>
    </StyledFooter>
  );
}

export default Footer;
