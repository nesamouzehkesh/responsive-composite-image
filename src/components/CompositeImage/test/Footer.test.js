import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Footer from '../Footer';

describe ('Test CompositeImage Footer component', () => {
  it ('renders Footer component normally and checks against snapshot', () => {
    const element = renderer
      .create (<Footer logo="test.png" title="Footer title" />)
      .toJSON ();

    expect (element).toMatchSnapshot ();
  });

  it ('renders Footer component normally with opacity and checks against snapshot', () => {
    const element = renderer
      .create (<Footer logo="test.png" title="Footer title" opacity="0.6" />)
      .toJSON ();

    expect (element).toMatchSnapshot ();
  });
});
