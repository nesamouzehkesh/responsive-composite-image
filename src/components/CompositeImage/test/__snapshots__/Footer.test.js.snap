// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CompositeImage Footer component renders Footer component normally and check with snapshot 1`] = `
<div
  className="sc-bdVaJa dfYFhE"
  opacity={undefined}
>
  <img
    className="sc-htpNat kbjeOc"
    src="test.png"
    title="Footer title"
  />
  <span
    className="sc-bwzfXH cyQaiT"
  >
    Footer title
  </span>
</div>
`;

exports[`Test CompositeImage Footer component renders Footer component normally with opacity and check with snapshot 1`] = `
<div
  className="sc-bdVaJa hmsgUZ"
  opacity="0.6"
>
  <img
    className="sc-htpNat kbjeOc"
    src="test.png"
    title="Footer title"
  />
  <span
    className="sc-bwzfXH cyQaiT"
  >
    Footer title
  </span>
</div>
`;
